### PR TITLE
Move some classes from spanner to core

### PR DESCRIPTION
**Pull request by vkedia**
Move Date and Timestamp to core
Removed ByteArrays and moved toBase64 and fromBase64 to ByteArray
Removed the fancy toString in ByteArrays since ByteArray.toString does a reasonable job.

### CODE REVIEW

**Comment by vkedia** [comment]
Closes #1769

**Comment by vkedia** [comment]
@garrettjonesgoogle Can you please take a look.

File: google-cloud-core/pom.xml [hunk 1]
```diff
@@ -116,6 +116,11 @@
       <version>3.0.0</version>
     </dependency>
     <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java-util</artifactId>
+      <version>3.0.0</version>
```
Comment (garrettjonesgoogle) [inline google-cloud-core/pom.xml]: I noticed that protobuf-java-util has a dependency on Guava. It doesn't expose Guava on its surface, but at the same time, it there isn't a shaded variant of it either. We are currently working towards providing a fully shaded variant of google-cloud-java soon (alongside the current variant where Guava isn't shaded).

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**